projects/vdk-plugins/vdk-jobs-troubleshooting/README.md

```diff
@@ -1 +1,245 @@
-### VDK-JOBS-TROUBLESHOOTING Plugin
+## VDK-JOBS-TROUBLESHOOTING Plugin
+
+The VDK JOB Troubleshooting plugin provides the ability to add various troubleshooting utilities which can be accessed
+during the data job runtime.
+
+Generally it's quite hard to produce a thread dump of a python process running inside a kubernetes pod. So this plugin
+provides a thread-dump utility
+
+### Adding the plugin to your data job
+
+Before you can use the plugin you should add it to your data job or custom-sdk, by adding the following line to your
+requirements.txt file:
+
+```commandline
+install_requires = vdk-jobs-troubleshooting
+```
+
+Next you need to add the list of utilities to use to your data job configuration. For example, to enable the thread-dump
+utility you have to add the following to your data job configuration.
+
+```commandline
+TROUBLESHOOT_UTILITIES_TO_USE=thread-dump
+```
+
+### Getting a thread dump from a running job
+
+During the startup of the data job the troubleshooting utility will log a message with the port it is running on. Example:
+
+```commandline
+Troubleshooting utility server will start on port 8783.
+```
+
+So in order to get a thread dump from the running data job do the following.
+
+1. Review the logs (using the kubectl logs pod/<data job pod it> command ) and find the troubleshooting utility port
+2. Start a port forward from your local machine to the target pod and port for example
+
+```commandline
+kubectl port-forward pods/starshot-prod-processing-csp-jira-1691419320-jcsn7 8783:8783
+```
+
+3. In a new/different console execute a curl command, to the pod. Example
+
+```commandline
+curl localhost:8783/threads
+```
+
+The thread-dump will be printed in the console.
```

Comment:
nit: Do we really need a complete thread-dump in the README? Also, was the output cleaned from internal information?

Reply:
fixed.